### ISSUE TITLE
Create PR for BIP44 support work for Bitcoinj

### ISSUE THREAD

**Issue by gary-rowe** [bitcoin-solutions/multibit-hd#441]
The BIP44 support work is currently held in our Bitcoinj-Alice fork. It would be better if this work could be generalised and pushed upstream for other projects to benefit from.

This is a complex change and may require a lot of support work before a PR would be accepted upstream. For this reason it has to occur after the first release of the project.

**Comment by devrandom** [comment]
I have a general implementation that allows a DKC to have an arbitrary account derivation path.  Should I'll compare my implementation to yours and see which one makes more sense to create a PR out of?

**Comment by jim618** [comment]
It sounds similar.
I found the serialisiing and deserialising of the wallets to be the least general part of the bitcoinj-alice code - not really good enough for a general purpose library.

**Comment by devrandom** [comment]
Yes, I had the same experience, due to the use of static functions.  Ended up with some factory classes to generalize.

**Comment by devrandom** [comment]
Created https://github.com/bitcoinj/bitcoinj/pull/339
